src/App.jsx

```diff
@@ -14,45 +16,109 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_NAME = 'name';
+const SORT_FIELD_LENGTH = 'length';
+
+function getPreparedGoods(goods, { sortField, isReversed }) {
+  let preparedGoods = [...goods];
+
+  if (sortField) {
+    switch (sortField) {
+      case SORT_FIELD_NAME:
+        preparedGoods.sort((good1, good2) => good1.localeCompare(good2));
+        break;
+
+      case SORT_FIELD_LENGTH:
+        preparedGoods
+          .sort((good1, good2) => good1[sortField] - good2[sortField]);
+        break;
+
+      default:
+        return 0;
+    }
+  }
+
+  if (isReversed) {
+    preparedGoods = preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setReverseField] = useState(false);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const visibleGoods = getPreparedGoods(
+    goodsFromServer,
+    { sortField, isReversed },
+  );
+
+  const resetSortingSettings = () => {
+    setSortField('');
+    setReverseField(false);
+  };
+
+  const isSortByReversed = (sortField || isReversed);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SORT_FIELD_NAME,
+          })}
```

Comment:
```suggestion
          className={cn(
            'button',
            'is-info',
            { 'is-light': sortField !== SORT_FIELD_NAME }
          )}
```
Let's improve readability, fix in all places